docs/reference/release-notes/highlights.asciidoc

```diff
@@ -18,6 +18,30 @@ For detailed information about this release, see the <<es-release-notes>> and
 // === Heading
 //
 // Description.
+
+[discrete]
+=== Faster indexing and searches on indexed `dense_vector` fields
+
+We've done several optimizations for indexing and searches on high-dimensional
+vectors (e.g. making HNSW graph hierarchical, speeding up graph merges,
+reusing data structures, better encoding of doc Ids). Lucene-level benchmarks
+reported up to 30% improvement in index throughput and 10% faster nearest
+neighbor searches on these fields. {es} indices that use indexed `dense_vector`
+fields should see these improvements.
+
+[discrete]
+===  Faster "count" searches for Points
```

Comment:
The count search optimizations that were introduced do not apply to Elasticsearch until we start leveraging the new `Weight#count` API. However there are speedups for range queries indeed, maybe we could check the sort of improvement that we observed on nightly benchmarks to share a more catchy headline such as "Up to 20% faster range queries". For instance I'm looking at https://elasticsearch-benchmarks.elastic.co/index.html#tracks/noaa/nightly/default/90d where `nightly-noaa-add-defaults-range_field_big_range-latency` looks like it got a ~35% speedup on 2022-02-22.